cloudinit/sources/helpers/azure.py

```diff
@@ -258,6 +258,51 @@ def _get_dhcp_endpoint_option_name():
     return azure_endpoint
 
 
+@azure_ds_telemetry_reporter
+def http_with_retries(url, **kwargs) -> str:
+    exc = None
```

Comment:
Might be worth a small docstring describing the intent here being periodic custom telemetry logging which readurl doesn't provide.

Reply:
Done. https://github.com/canonical/cloud-init/pull/619/commits/8dfe0598fc65e1aae13fb6641f7cba7eae918a70